ta/pkcs11/src/pkcs11_helpers.c

```diff
@@ -64,7 +64,86 @@ static const struct any_id __maybe_unused string_ta_cmd[] = {
 	PKCS11_ID(PKCS11_CMD_PING),
 };
 
+static const struct any_id __maybe_unused string_rc[] = {
+	PKCS11_ID(PKCS11_CKR_OK),
+	PKCS11_ID(PKCS11_CKR_GENERAL_ERROR),
+	PKCS11_ID(PKCS11_CKR_DEVICE_MEMORY),
+	PKCS11_ID(PKCS11_CKR_ARGUMENTS_BAD),
+	PKCS11_ID(PKCS11_CKR_BUFFER_TOO_SMALL),
+	PKCS11_ID(PKCS11_CKR_FUNCTION_FAILED),
+	PKCS11_ID(PKCS11_CKR_SIGNATURE_INVALID),
+	PKCS11_ID(PKCS11_CKR_ATTRIBUTE_TYPE_INVALID),
+	PKCS11_ID(PKCS11_CKR_ATTRIBUTE_VALUE_INVALID),
+	PKCS11_ID(PKCS11_CKR_OBJECT_HANDLE_INVALID),
+	PKCS11_ID(PKCS11_CKR_KEY_HANDLE_INVALID),
+	PKCS11_ID(PKCS11_CKR_MECHANISM_INVALID),
+	PKCS11_ID(PKCS11_CKR_SESSION_HANDLE_INVALID),
+	PKCS11_ID(PKCS11_CKR_SLOT_ID_INVALID),
+	PKCS11_ID(PKCS11_CKR_MECHANISM_PARAM_INVALID),
+	PKCS11_ID(PKCS11_CKR_TEMPLATE_INCONSISTENT),
+	PKCS11_ID(PKCS11_CKR_TEMPLATE_INCOMPLETE),
+	PKCS11_ID(PKCS11_CKR_PIN_INCORRECT),
+	PKCS11_ID(PKCS11_CKR_PIN_LOCKED),
+	PKCS11_ID(PKCS11_CKR_PIN_EXPIRED),
+	PKCS11_ID(PKCS11_CKR_PIN_INVALID),
+	PKCS11_ID(PKCS11_CKR_PIN_LEN_RANGE),
+	PKCS11_ID(PKCS11_CKR_SESSION_EXISTS),
+	PKCS11_ID(PKCS11_CKR_SESSION_READ_ONLY),
+	PKCS11_ID(PKCS11_CKR_SESSION_READ_WRITE_SO_EXISTS),
+	PKCS11_ID(PKCS11_CKR_OPERATION_ACTIVE),
+	PKCS11_ID(PKCS11_CKR_KEY_FUNCTION_NOT_PERMITTED),
+	PKCS11_ID(PKCS11_CKR_OPERATION_NOT_INITIALIZED),
+	PKCS11_ID(PKCS11_CKR_TOKEN_WRITE_PROTECTED),
+	PKCS11_ID(PKCS11_CKR_TOKEN_NOT_PRESENT),
+	PKCS11_ID(PKCS11_CKR_TOKEN_NOT_RECOGNIZED),
+	PKCS11_ID(PKCS11_CKR_ACTION_PROHIBITED),
+	PKCS11_ID(PKCS11_CKR_ATTRIBUTE_READ_ONLY),
+	PKCS11_ID(PKCS11_CKR_PIN_TOO_WEAK),
+	PKCS11_ID(PKCS11_CKR_CURVE_NOT_SUPPORTED),
+	PKCS11_ID(PKCS11_CKR_DOMAIN_PARAMS_INVALID),
+	PKCS11_ID(PKCS11_CKR_USER_ALREADY_LOGGED_IN),
+	PKCS11_ID(PKCS11_CKR_USER_ANOTHER_ALREADY_LOGGED_IN),
+	PKCS11_ID(PKCS11_CKR_USER_NOT_LOGGED_IN),
+	PKCS11_ID(PKCS11_CKR_USER_PIN_NOT_INITIALIZED),
+	PKCS11_ID(PKCS11_CKR_USER_TOO_MANY_TYPES),
+	PKCS11_ID(PKCS11_CKR_USER_TYPE_INVALID),
+	PKCS11_ID(PKCS11_CKR_SESSION_READ_ONLY_EXISTS),
+	PKCS11_ID(PKCS11_RV_NOT_FOUND),
+	PKCS11_ID(PKCS11_RV_NOT_IMPLEMENTED),
+};
+
+/*
+ * Conversion between PKCS11 TA and GPD TEE return codes
+ */
+uint32_t tee2pkcs_error(TEE_Result res)
```

Comment:
Is this useful?

Reply:
Well, it will be used later... 
I'll remove this. We'll see that when needed.
Thanks.

Reply:
OK